cmd/pr/approvealign/runner.go

```diff
@@ -0,0 +1,121 @@
+package approvealign
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	"github.com/google/go-github/v72/github"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/giantswarm/devctl/v7/pkg/githubclient"
+	// "golang.org/x/oauth2" // No longer needed if using ghClient.GetUnderlyingClient
+)
+
+const (
+	githubTokenEnvVar = "GITHUB_TOKEN" // Standard environment variable for GitHub token
+)
+
+type runner struct {
+	flag   *flag
+	logger *logrus.Logger
+	stdout io.Writer
+	stderr io.Writer
+}
+
+func (r *runner) Run(cmd *cobra.Command, args []string) error {
+	ctx := context.Background()
+	if err := r.flag.Validate(); err != nil {
+		return microerror.Mask(err)
+	}
+	return r.run(ctx, cmd, args)
+}
+
+func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) error {
+	r.logger.Debug("running approvealign command")
+
+	fmt.Fprintln(r.stdout, "Auto-approving all 'Align files' PRs that have passing status checks...")
+
+	githubToken, found := os.LookupEnv(githubTokenEnvVar)
+	if !found {
+		return microerror.Maskf(executionFailedError, "environment variable %#q not found, please set it to your GitHub personal access token", githubTokenEnvVar)
+	}
+
+	ghClientService, err := githubclient.New(githubclient.Config{
+		Logger:      r.logger,
+		AccessToken: githubToken,
+	})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// This method ghClientService.GetUnderlyingClient(ctx) will be made public in pkg/githubclient/client.go
+	trueUnderlyingClient := ghClientService.GetUnderlyingClient(ctx)
```

Comment:
Might be better to rename this something more descriptive like `githubClient` ? Otherwise it's not clean _what_ it is underlying :)